chore(curriculum): add transcripts to working with file systems lecture (#58260)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/lecture-working-with-file-systems/672ac39a22b0cc30eb8fd65e.md

Watch the lecture video and answer the questions below.

# --transcript--

What are best practices for file and folder organization in web applications?

> When building out web applications, it is important to think about the organization of your files and folders. This will help keep your code organized and make it easier to maintain.

Author: Why remove that?

Let's take a look at an example folder structure for an HTML and CSS project:

```bash
.
├── /assets
│   ├── /images
│   │   ├── logo.png
│   │   ├── banner.jpg
│   │   └── icons.svg
│   ├── /fonts
│   │   ├── custom-font.woff
│   │   └── custom-font.woff2
├── /css
│   ├── main.css
│   ├── about.css
│   └── contact.css
├── index.html
├── about.html
├── contact.html
└── README.md
```

> At the top of the example, we have a single dot. That dot represents what is known as the root directory.

Reviewer: ```suggestion
At the top of the example, there is a single dot, which represents the root directory.
```

> The root directory is the top-level directory in a file system. It is the starting point for all file paths and contains all other directories and files within it. Directory is another name for a folder.

Reviewer: ```suggestion
The root directory is the top-level directory in a file system, serving as the starting point for all file paths and containing all other directories and files. A directory is another name for a folder.
```

Inside the root directory, the first directory shown is the `assets` directory. The term `assets` refers to any files that are used in the project, such as images, fonts, or other resources.

In this example, there are two directories inside the `assets` directory called `images` and `fonts`.

The next directory would be the `css` directory. Sometimes this is also referred to as a stylesheets directory. This is where you would store all of your CSS files.

> Below the `css` directory, you have your HTML files. The `index.html` represents the homepage, while the `about.html` and `contact.html` represent the about and contact pages, respectively.

Reviewer: ```suggestion
Below the `css` directory are the HTML files. The `index.html` file represents the homepage, while `about.html` and `contact.html` represent the About and Contact pages, respectively.
```
